recipe/build-pyarrow.sh

```diff
@@ -38,6 +38,12 @@ if [[ "${target_platform}" == "linux-aarch64" ]]; then
     export PYARROW_CMAKE_OPTIONS="-DARROW_ARMV8_ARCH=armv8-a ${PYARROW_CMAKE_OPTIONS}"
 fi
 
+# Limit number of threads used to avoid hardware oversubscription
+if [[ "${target_platform}" == "linux-aarch64" ]] || [[ "${target_platform}" == "linux-ppc64le" ]]; then
+     export PYARROW_CMAKE_OPTIONS="${PYARROW_CMAKE_OPTIONS} -DCMAKE_BUILD_PARALLEL_LEVEL=4"
+fi
```

Comment:
Looks like [CI didn't use it]( https://app.travis-ci.com/github/conda-forge/arrow-cpp-feedstock/jobs/568325757#L3660-L3663 ). Will try an environment variable, but not sure if that will work.

Reply:
Environment variable seems to work. Trying in C++ builds as well.